dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java

```diff
@@ -436,18 +425,7 @@ public T getExtension(String name, boolean wrap) {
         if ("true".equals(name)) {
             return getDefaultExtension();
         }
-        final Holder<Object> holder = getOrCreateHolder(name);
-        Object instance = holder.get();
-        if (instance == null) {
-            synchronized (holder) {
-                instance = holder.get();
-                if (instance == null) {
-                    instance = createExtension(name, wrap);
-                    holder.set(instance);
-                }
-            }
-        }
-        return (T) instance;
+        return cachedInstances.computeIfAbsent(name, (k) -> createExtension(k, wrap));
```

Comment:
I'm worried this might create a lot of class objects because of the lambda.

Reply:
Yes, Using a get-null prediction could be better.

Reply:
rollback will make this action not atomic

Reply:
The method computeIfAbsent of ConcurrentHashMap is atomic. That function is applied at most once per key.

Reply:
but lambda might create a lot of class object, and will lead to metadata space increase

Reply:
Yes，lambda capture the context variable of wrap as closure,so It renew a class for every invocation.